woocommerce/woocommerce-gutenberg-products-block
# Create Cross-Sells product list

Fixes #6558

#### Accessibility


#### Other Checks

- [ ] This PR adds/removes a feature flag & I've updated [this doc](../docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [x] This PR adds/removes an experimental interfaces and I've updated [this doc](../docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).

### Screenshots

<table>
<tr>
<td>Before:
<br><br>

![6558-before](https://user-images.githubusercontent.com/3323310/176687090-dcbff95e-456f-489f-ab0c-2dddb1603c3d.png)
</td>
<td>After:
<br><br>

![6558-after](https://user-images.githubusercontent.com/3323310/176687105-29613d0a-1baa-4475-a9cb-24624aae3476.png)
</td>
</tr>
</table>

### Testing

#### Test editor behaviour when adding Cart block

1. Create a test page and add the Cart block to it.
2. Verify that the Cart block contains the Cart Cross-Sells inner block.
3. Verify that the Cart Cross-Sells inner block contains the Heading and Cart Cross-Sells products inner blocks.
4. Select the Cart Cross-Sells products inner block and verify that the number of visible Cross-Sells products can be adjusted in the editor.

#### Test editor behaviour after removing Cross-Sells block

1. Create a test page and add the Cart block to it.
2. Verify that the Cart block contains the Cart Cross-Sells inner block.
3. Delete the Cart Cross-Sells inner block.
4. Verify that the Cart Cross-Sells inner block can be reinserted.

#### Test frontend behaviour if cart items does not have Cross-Sells products

1. Visit the frontend.
2. Add a product to the cart that does not Cross-Sells products.
3. Go to the Cart page.
4. Verify that the Cross-Sells product block is not visible.

#### Test frontend behaviour if cart items have Cross-Sells products

1. Visit the frontend.
2. Add a product to the cart that has Cross-Sells products.
3. Go to the Cart page.
4. Verify that the Cross-Sells product block is visible.
5. Add one of the Cross-Sells products, which also has Cross-Sells products, to the cart.
6. Verify that the added Cross-Sells product had been added to the cart and is no longer visible as a Cross-Sells product.
7. Verify that the Cross-Sells products, of the product that had been added to the cart in step 6., are visible in the Cross-Sells block.

#### Test frontend behaviour if cart items have a Cross-Sells product that is out of stock

1. Visit the frontend.
2. Add a product to the cart that has a Cross-Sells product which is out of stock.
3. Go to the Cart page.
4. Verify that the Cross-Sells product block is visible.
5. Verify that the Cross-Sells product, which is out of stock, does not show an `Add to Cart` button, but a `Read More` button, that links to the corresponding product page.

#### Test number of visible Cross-Sells products in the frontend.

1. Ensure that you have one test product which has more than 6 Cross-Sells products.
2. Open the cart page in the editor and set the number of visible Cross-Sells items to 1.
3. Go to the frontend and add the test product from step 1 to the cart.
4. Go to the cart and verify that only 1 Cross-Sells product is visible.
5. Repeat steps 2 until 4 and set the number of visible Cross-Sells items to 2, 3, 4, 5 & 6.


### WooCommerce Visibility




### Performance Impact

n/a

### Changelog

> Create Cross-Sells product list for showing the Cross-Sells products on the Cart block.
